format/bplist/bplist.go

```diff
@@ -191,8 +210,26 @@ type trailer struct {
 }
 
 type plist struct {
-	t trailer
-	o []uint64
+	t           trailer
+	o           []uint64
+	objectStack []uint64
+}
+
+func (pl *plist) pushIndex(idx uint64) {
+	pl.objectStack = append(pl.objectStack, idx)
+}
+
+func (pl *plist) popIndex() {
+	pl.objectStack = pl.objectStack[:len(pl.objectStack)-1]
+}
+
+func (pl *plist) indexIsInStack(idx uint64) bool {
+	for _, existing := range pl.objectStack {
+		if existing == idx {
+			return true
+		}
+	}
+	return false
```

Comment:
Nice 👍  i should probably do something like this in other places were i just have some kind of depth limit instead

Reply:
I though about using a map instead of a slice for looking up the previously decoded references but figured a slice would be faster for the common case where things aren't nested too deeply

Reply:
Yeap slice is probably fine. I've noticed that fq currently allocates quite a lot (each field is a couple allocs at least i think) so that is usually what shows up in both cpu and memory profiles reports.